### PR TITLE
Enable rubocop for entire `forest-iac` repo

### LINKED ISSUE #82: Rubocop is not working outside scripts/
**Issue summary**

`rubocop` is not working outside `scripts/` directory as defined in:
https://github.com/ChainSafe/forest-iac/blob/5c00f05f380ad445a2f159538483c08c0681d930/.github/workflows/scripts-lint.yml#L29-L32

Some of the Ruby code was moved to other directories and has since strived away from the laws enforced by the dreaded Ruby linter.

Enable Rubocop everywhere and fix the offending issues.

**Other information and links**

### PR DESCRIPTION
**Pull request by jdjaustin**
**Summary of changes**
Changes introduced in this pull request:
- Fix rubocop lint errors in existing scripts outside of `scripts/` directory and enable rubocop lint for entire `forest-iac` repository.

**Reference issue to close (if applicable)**

Closes #82 

**Other information and links**

### CODE REVIEW

**Comment by aatifsyed** [comment]
@jdjaustin why is this open rather than a `Draft`?

File: terraform/modules/daily_snapshot/service/list_snapshots.rb [hunk 1]
```diff
@@ -12,23 +12,33 @@
...
-    @url = url
+    raise ArgumentError,
+          'Missing argument'
+  end
+
+  def check_network_param(params)
+    return unless params[:network].downcase != 'mainnet' && params[:network].downcase != 'calibnet'
+
+    raise ArgumentError,
+          'Invalid network'
```
Comment (LesnyRumcajs) [inline terraform/modules/daily_snapshot/service/list_snapshots.rb:41]: It'd be useful to print the name of the network that is not valid, for debugging purposes

File: terraform/modules/daily_snapshot/service/list_snapshots.rb [hunk 2]
```diff
@@ -46,26 +56,52 @@ def delete
...
+end
+
+def prepare_to_update_snapshot_list
+  snapshot_format = \
+    /^(?:[^_]+?)_snapshot_(?<network>[^_]+?)_(?<date>\d{4}-\d{2}-\d{2})_height_(?<height>\d+)(?:\.forest)?\.car.zst$/
   snapshot_list = []
+  [snapshot_format, snapshot_list]
+end
+
+def match_assignments(bucket, chain_name, endpoint, file, match)
```
Comment (LesnyRumcajs) [inline terraform/modules/daily_snapshot/service/list_snapshots.rb:74]: Some documentation for the methods would do, not only this but the other ones (which are doing non-obvious stuff) as well

File: terraform/modules/daily_snapshot/service/list_snapshots.rb [hunk 3]
```diff
@@ -46,26 +56,52 @@ def delete
...
-  snapshot_format = /^([^_]+?)_snapshot_(?<network>[^_]+?)_(?<date>\d{4}-\d{2}-\d{2})_height_(?<height>\d+)(\.forest)?\.car.zst$/
+def prepare_to_list_snapshots(chain_name, bucket)
+  ls_format = %r{\d{4}-\d{2}-\d{2} \d{2}:\d{2}\s*\d*\s*s3://#{bucket}/#{chain_name}/(.+)}
 
   output = `s3cmd ls s3://#{bucket}/#{chain_name}/`
 
+  [ls_format, output]
+end
+
+def prepare_to_update_snapshot_list
```
Comment (LesnyRumcajs) [inline terraform/modules/daily_snapshot/service/list_snapshots.rb:67]: When introducing new methods, it'd be interesting to grab some inspiration from the existing `snapshots_prune_spec.rb` test and write some unit tests. Unit tests are valuable, but in the world of interpreted code, they are the only thing that keeps the application from runtime errors.

File: terraform/modules/daily_snapshot/service/tests/snapshots_prune_spec.rb [hunk 4]
```diff
@@ -22,75 +22,124 @@ def create_consecutive_days_snapshots(start_date, count)
   end
 end
 
-describe 'prune_snapshots' do
+# Helper function for `prune_snapshots`
```
Comment (LesnyRumcajs) [inline terraform/modules/daily_snapshot/service/tests/snapshots_prune_spec.rb:25]: This description is not really helpful

**Review by LesnyRumcajs — COMMENTED** [review]
Thanks for taking care of the issues! I'd say some changes are needed to make it nicer, see the comments. Not requesting changes because I don't want to block this PR during my PTO.

File: terraform/modules/daily_snapshot/service/list_snapshots.rb [hunk 5]
```diff
@@ -12,23 +12,33 @@
...
 
-    @network = network
-    @date = date.to_date
-    @height = height.to_i
-    @file_name = file_name
-    @url_s3 = url_s3
-    @url = url
+    raise ArgumentError,
+          'Missing argument'
+  end
```
[the remark this answers is not in the document]
Comment (jdjaustin) [reply terraform/modules/daily_snapshot/service/list_snapshots.rb:35]: IIRC the current format was suggested by Rubocop, but I can see about changing it.

[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply terraform/modules/daily_snapshot/service/list_snapshots.rb:35]: let's use what rubocop suggests then, in rubocop we trust

**Comment by jdjaustin** [comment]
The benchmark script has been retired, and IIUC from the discussions in Slack, the team is planning to move away from using Ruby in `forest-iac` long-term, so I'm closing this PR for now. If this becomes a priority in the future, we can reopen if necessary.

**Comment by LesnyRumcajs** [comment]
> The benchmark script has been retired, and IIUC from the discussions in Slack, the team is planning to move away from using Ruby in `forest-iac` long-term, so I'm closing this PR for now. If this becomes a priority in the future, we can reopen if necessary.

I don't think there's an immediate plan to move the existing services to another technology (given they are working just fine). That said, I enabled it and fixed it in another PR https://github.com/ChainSafe/forest-iac/pull/388

**Pull request closed without merging**